### DISCUSSION TITLE
diesel insert mysql

### DISCUSSION THREAD

**Discussion by jutbin** [diesel-rs/diesel#4147] [Q&A]
How to insert date and time with time zone in diesel mysql

**Comment by weiznich** [comment]
Duplicate of https://github.com/diesel-rs/diesel/discussions/4143#discussioncomment-10239898.

Please don't open any duplicate already answered questions anymore.

**Reply by jutbin** [reply]
this not duplicate questions

**Reply by jutbin** [reply]
I use diesel to add data in MySQL. The field is of type time::OffsetDateTime. The time field type in MySQL is timestamp. However, when compiling, I get an error message: the trait `diesel::Expression` is not implemented for `OffsetDateTime`, which is required by `OffsetDateTime: AsExpression<diesel::sql_types::Timestamp>`. MySQL's timestamp supports time zones. Why can't diesel work?

**Reply by weiznich** [reply]
> this not duplicate questions

It's literally the same question you asked before in the other thread. You already got an answer there. There is no reason to ask the same question again just because you did not like the answer.

> MySQL's timestamp supports time zones. 

Please provide links to the MySQL documentation for a `TIMESTAMP WITH TIMEZONE` type, otherwise you just claim things.
